pkg/types/interfacetests/utils.go

```diff
@@ -36,12 +41,91 @@ func runTests[T TestingT[T]](t T, tester BasicTester[T], tests []testcase[T]) {
 	}
 }
 
+// Batch chain write takes a batch call entry and writes it to the chain using the ChainWriter.
+func batchChainWrite[T TestingT[T]](t T, tester ChainComponentsInterfaceTester[T], batchCallEntry BatchCallEntry, mockRun bool) {
+	// This is necessary because the mock helper function requires the entire batchCallEntry rather than an individual testStruct
+	if mockRun {
+		cw := tester.GetChainWriter(t)
+		err := cw.SubmitTransaction(tests.Context(t), AnyContractName, "batchChainWrite", batchCallEntry, "", "", nil, big.NewInt(0))
+		require.NoError(t, err)
+		return
+	}
+	nameToAddress := make(map[string]string)
+	boundContracts := tester.GetBindings(t)
+	for _, bc := range boundContracts {
+		nameToAddress[bc.Name] = bc.Address
+	}
+
+	// For each contract in the batch call entry, submit the read entries to the chain
+	for contractName, contractBatch := range batchCallEntry {
+		require.Contains(t, nameToAddress, contractName)
+		for _, readEntry := range contractBatch {
+			val, isOk := readEntry.ReturnValue.(*TestStruct)
+			if !isOk {
+				require.Fail(t, "expected *TestStruct for contract: %s read: %s, but received %T", contractName, readEntry.Name, readEntry.ReturnValue)
+			}
+			SubmitTransactionToCW(t, tester, "addTestStruct", val, types.BoundContract{Name: contractName, Address: nameToAddress[contractName]}, types.Unconfirmed)
+		}
+	}
+}
+
+// SubmitTransactionToCW submits a transaction to the ChainWriter and waits for it to reach the given status.
+func SubmitTransactionToCW[T TestingT[T]](t T, tester ChainComponentsInterfaceTester[T], method string, args any, contract types.BoundContract, status types.TransactionStatus) string {
+	tester.DirtyContracts()
```

Comment:
Since each test case sets up new contracts now can we remove this?

Reply:
I actually added this back in to ensure not every tests created new contracts. i.e. the ones that don't send any transactions don't need new contracts to be deployed after them. This allows for a slight optimization in runtime / token spend on testnet.